pkg/image/admission/imagepolicy/api/v1/types.go

```diff
@@ -0,0 +1,138 @@
+package v1
+
+import (
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
+)
+
+// ImagePolicyConfig is the configuration for control of images running on the platform.
+type ImagePolicyConfig struct {
+	unversioned.TypeMeta `json:",inline"`
+
+	// ExecutionRules determine whether the use of an image is allowed in an object with a pod spec.
+	// By default, these rules only apply to pods, but may be extended to other resource types.
+	// If all execution rules are negations, the default behavior is allow all. If any execution rule
+	// is an allow, the default behavior is to reject all.
+	ExecutionRules []ImageExecutionPolicyRule `json:"executionRules"`
+
+	// TODO: enable additional rule types
+	// ConsumptionRules are applied when creating resources with pod specs.
+	// ConsumptionRules []ImageConsumptionPolicyRule `json:"consumptionRules"`
+	// PlacementRules are applied when creating resources with pod specs.
+	// PlacementRules []ImagePlacementPolicyRule `json:"placementRules"`
+}
+
+// ImageExecutionPolicyRule determines whether a provided image may be used on the platform.
+type ImageExecutionPolicyRule struct {
+	ImageCondition `json:",inline"`
+
+	// Resolve indicates that images referenced by this resource must be resolved
+	Resolve bool `json:"resolve"`
+}
+
+// ImageCondition defines the conditions for matching a particular image source. The conditions below
+// are all required (logical AND). If Reject is specified, the condition is false if all conditions match,
+// and true otherwise.
+type ImageCondition struct {
+	// Name is the name of this policy rule for reference. It must be unique across all rules.
+	Name string `json:"name"`
+	// IgnoreNamespaceOverride prevents this condition from being overriden when the
+	// `alpha.image.policy.openshift.io/ignore-rules` is set on a namespace and contains this rule name.
+	IgnoreNamespaceOverride bool `json:"ignoreNamespaceOverride"`
+
+	// Reject indicates this rule is inverted - if all conditions below match, then this rule is considered to
+	// not match.
+	Reject bool `json:"reject"`
```

Comment:
I think I want to be able to invert the match here and still specify `Reject` in the `ImageExecutionPolicyRule`.  That allows me to have a veto rule based on an inverted match.

```go
ImageExecutionPolicyRule.Reject = true
ImageExecutionPolicyRule.Invert = true
ImageExecutionPolicyRule.MatchRegistries = []string{"foo"}
```

I think that rule says, "if you don't match the registry foo, reject".  That is different than "if you match the registry foo, accept" since the first rule allows me to specify the rejections I want and default allow instead of having to enumerate all allows.